packages/venia-ui/lib/components/CheckoutPage/ShippingMethod/editForm.js

```diff
@@ -0,0 +1,69 @@
+import React, { useMemo } from 'react';
+import { Form } from 'informed';
+
+import { mergeClasses } from '../../../classify';
+import Button from '../../Button';
+import ShippingRadios from './shippingRadios';
+import defaultClasses from './editForm.css';
+
+export const modes = {
+    INITIAL: 'initial',
+    UPDATE: 'update'
+};
+
+const EditForm = props => {
+    const {
+        handleCancelUpdate,
+        handleSubmit,
+        isLoading,
+        mode,
+        pageIsUpdating,
+        selectedShippingMethod,
+        shippingMethods
+    } = props;
+
+    const initialButtonDisabled = pageIsUpdating || !shippingMethods.length;
+    const classes = mergeClasses(defaultClasses, props.classes);
+
+    const buttons = useMemo(() => {
+        if (mode === modes.INITIAL) {
+            if (isLoading) return null;
+
+            return (
+                <div className={classes.initialButtonContainer}>
+                    <Button
+                        priority="normal"
+                        type="submit"
+                        disabled={initialButtonDisabled}
+                    >
+                        {'Continue to Payment Information'}
+                    </Button>
+                </div>
+            );
+        }
+
+        if (mode === modes.UPDATE) {
+            return (
+                <div className={classes.updateButtonContainer}>
+                    <Button onClick={handleCancelUpdate}>{'Cancel'}</Button>
+                    <Button priority="high" type="submit">
```

Comment:
You should disable this button while the update is in flight.

Reply:
Good call. Resolved in https://github.com/magento/pwa-studio/pull/2280/commits/3e47f4f62fdb55800ecd72a106b749f94b20f3e0.